core/src/main/scala/chisel3/internal/firrtl/Converter.scala

```diff
@@ -344,13 +346,16 @@ private[chisel3] object Converter {
       }
     // extract underlying type for const
     case d if (checkConst && d.isConst) => fir.ConstType(extractType(d, clearDir, info, checkProbe, false))
-    case _: Clock      => fir.ClockType
-    case _: AsyncReset => fir.AsyncResetType
-    case _: ResetType  => fir.ResetType
-    case d: EnumType   => fir.UIntType(convert(d.width))
-    case d: UInt       => fir.UIntType(convert(d.width))
-    case d: SInt       => fir.SIntType(convert(d.width))
-    case d: Analog => fir.AnalogType(convert(d.width))
+    case _: Clock                                 => fir.ClockType
+    case _: AsyncReset                            => fir.AsyncResetType
+    case _: ResetType                             => fir.ResetType
+    case d: EnumType                              => fir.UIntType(convert(d.width))
+    case d: UInt                                  => fir.UIntType(convert(d.width))
+    case d: SInt                                  => fir.SIntType(convert(d.width))
+    case d: Analog                                => fir.AnalogType(convert(d.width))
+    case d: Property[_] if d.tpe =:= typeOf[Int]  => fir.IntegerPropertyType
+    case d: Property[_] if d.tpe =:= typeOf[Long] => fir.IntegerPropertyType
+    case d: Property[_] if d.tpe =:= typeOf[BigInt] => fir.IntegerPropertyType
```

Comment:
I put a type tag into `Property`, so I could do this check and conversion to the appropriate firrtl Type here. This seemed like the least amount of ceremony, but I would prefer more exhaustiveness checking. Another approach might be to add a method on `PropertyType` to get a case object from some sealed hierarchy we control, which is then matched and converted to a Type here. At least that way, the match here can be checked by the compiler for exhaustiveness, and it move around the ceremony for adding new types, but doesn't make it much more verbose. Curious what others think.